marquez/airflow.py

```diff
@@ -1,93 +1,84 @@
 import json
 import pendulum
-import airflow.models
+from airflow.models import DAG, Log
 from airflow.utils.db import provide_session
 from marquez_client.marquez import MarquezClient
 
+from marquez.utils import JobIdMapping
 
-class MarquezDag(airflow.models.DAG):
+
+class MarquezDag(DAG):
+    _jobIdMapping = None
+    _mqz_client = None
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        self.mqz_client = MarquezClient()
         self.mqz_namespace = kwargs['default_args'].get('mqz_namespace', 'unknown')
         self.mqz_location = kwargs['default_args'].get('mqz_location', 'unknown')
         self.mqz_input_datasets = kwargs['default_args'].get('mqz_input_datasets', [])
         self.mqz_output_datasets = kwargs['default_args'].get('mqz_output_datasets', [])
+        self._jobIdMapping = JobIdMapping()
 
     def create_dagrun(self, *args, **kwargs):
-        job_name = self.dag_id
-        job_run_args = "{}"                     # TODO retrieve from DAG/tasks
-        start_time = pendulum.instance(kwargs['execution_date']).to_datetime_string()
-        end_time = None
-
-        self.mqz_client.set_namespace(self.mqz_namespace)
-        self.mqz_client.create_job(job_name, self.mqz_location, self.mqz_input_datasets, self.mqz_output_datasets,
-                                   self.description)
-        mqz_job_run_id = self.mqz_client.create_job_run(job_name, job_run_args=job_run_args,
-                                                        nominal_start_time=start_time,
-                                                        nominal_end_time=end_time).run_id
-        self.mqz_client.mark_job_run_running(mqz_job_run_id)
-
-        self.marquez_log('job_running', json.dumps(
-            {'namespace': self.mqz_namespace,
-             'name': job_name,
-             'description': self.description,
-             'location': self.mqz_location,
-             'runArgs': job_run_args,
-             'nominal_start_time': start_time,
-             'nominal_end_time': end_time,
-             'jobrun_id': mqz_job_run_id,
-             'inputDatasetUrns': self.mqz_input_datasets,
-             'outputDatasetUrns': self.mqz_output_datasets
-             }))
-
-        run = super().create_dagrun(*args, **kwargs)
-        airflow.models.Variable.set(run.run_id, mqz_job_run_id)
-
+        run_args = "{}"  # TODO extract the run Args from the tasks
```

Comment:
Is this something we still want to do?

Reply:
I'm not sure actually. Right now we are just hardcoding that.

Reply:
Yeah, collecting the run args will assist in debugging job failures, etc. So, it's something we very much want to capture and still need to discuss (i.e. stringified json (=current approach) vs map vs array)

Reply:
I agree with the debugging part, we just need to identify what run_args are in Airflow-land (is it args used by the operators, or the Dag, etc..)